ddtrace/llmobs/_integrations/google_genai.py

```diff
@@ -16,3 +49,147 @@ def _set_base_span_tags(
+
+        span._set_ctx_items(
+            {
+                SPAN_KIND: "llm",
+                MODEL_NAME: model_name,
+                MODEL_PROVIDER: provider_name,
+                METADATA: self._extract_metadata(config),
+                INPUT_MESSAGES: self._extract_input_message(args, kwargs, config),
+                OUTPUT_MESSAGES: self._extract_output_message(response),
+                METRICS: extract_metrics_google_genai(response),
+            }
+        )
+
+    def _extract_message_from_part_google_genai(self, part, role):
+        """
+        part is a PartUnion = Union[File, Part, PIL_Image, str]
+        """
+        message = {"role": role}
+        if isinstance(part, str):
+            message["content"] = part
+            return message
+
+        # only one field is set in a Part
+        text = _get_attr(part, "text", None)
+        if text:
+            message["content"] = text
+            return message
+
+        function_call = _get_attr(part, "function_call", None)
+        if function_call:
+            function_call_dict = function_call.model_dump()
+            message["tool_calls"] = [
+                {"name": function_call_dict.get("name", ""), "arguments": function_call_dict.get("args", {})}
+            ]
+            return message
+
+        function_response = _get_attr(part, "function_response", None)
+        if function_response:
+            function_response_dict = function_response.model_dump()
+            message["content"] = "[tool result: {}]".format(function_response_dict.get("response", ""))
+            return message
+
+        executable_code = _get_attr(part, "executable_code", None)
+        if executable_code:
+            language = _get_attr(executable_code, "language", "UNKNOWN")
+            code = _get_attr(executable_code, "code", "")
+            message["content"] = "[executable code ({language}): {code}]".format(language=language, code=code)
+            return message
+
+        code_execution_result = _get_attr(part, "code_execution_result", None)
+        if code_execution_result:
+            outcome = _get_attr(code_execution_result, "outcome", "OUTCOME_UNSPECIFIED")
+            output = _get_attr(code_execution_result, "output", "")
+            message["content"] = "[code execution result ({outcome}): {output}]".format(outcome=outcome, output=output)
+            return message
+
+        thought = _get_attr(part, "thought", None)
+        # thought is just a boolean indicating if the part was a thought
+        if thought:
+            message["content"] = "[thought: {}]".format(thought)
+            return message
+
+        return {"content": "Unsupported file type: {}".format(type(part)), "role": role}
+
+    def _extract_input_message(self, args, kwargs, config):
+        messages = []
+        # system instruction in config
+        if config is not None:
+            system_instruction = _get_attr(config, "system_instruction", None)
```

Comment:
We don't need to check if `config is not None`. Let's just try grabbing the system_instruction directly via `_get_attr()` and avoid all these nested ifs